### PR TITLE
refactor documentation with snippets loading procedure section

### PR DESCRIPTION
**Pull request by SnowCone161803**
I have:
* Created a new section "4.1.3 How snippets are loaded"
* Moved documentation from "4.1 Adding Snippets" section into "4.1.3 How snippets are loaded"

### CODE REVIEW

**Comment by SirVer** [comment]
What triggered this change? Did you not find the documentation in the different spot?

**Comment by SirVer** [comment]
@SnowCone161803 Gordon, friendly ping?

**Comment by SnowCone161803** [comment]
Sorry! I missed the first message! Thanks for the ping!

I scanned the documentation for how the snippets were loaded and I started off with the [snippet search path documentation](https://github.com/SirVer/ultisnips/blob/master/doc/UltiSnips.txt#L368)

I had completely missed it in the [adding snippets](https://github.com/SirVer/ultisnips/blob/master/doc/UltiSnips.txt#L571) section as I thought that this was more basic creation of snippets.

I figured adding another section in the index might make it a little easier ;)

**Comment by SirVer** [comment]
@SnowCone161803 Your reasoning makes sense, it would be nice to surface clearer how snippets are loaded. But now the headers do not make sense anymore: 

`4.1.3 How snippets are loaded` introduces also the basic syntax (comments, extends, priority), but now comes after `snippet` and all the options.

I suggest the following changes:
- Rename 4: "syntax" to "Authoring snippets".
- Rename 4.1: "Adding snippets" to "Introduction" or "Basics"
- Move 3.3 "Snippet Search Path" into a new section  "4.1.1. How snippets are loaded".

**Comment by SnowCone161803** [comment]
Changes made.

I realised that after I pushed the changes, then squashed the commits that I had started up Travis twice instead of once. I think the way that I would normally work on my own projects would cause this to happen lots. Could this cause any issues?

File: doc/UltiSnips.txt [hunk 1]
```diff
@@ -1820,8 +1810,9 @@ https://github.com/sirver/ultisnips/issues
 8. Contributors                                      *UltiSnips-contributors*
 
 UltiSnips has been started and maintained from Jun 2009 - Dec 2015 by Holger
-Rapp (@SirVer, SirVer@gmx.de). It is now maintained by Stanislav Seletskiy
-(@seletskiy).
+Rapp (@SirVer, SirVer@gmx.de). Up to April 2018 it was maintained by Stanislav
+Seletskiy (@seletskiy). Now, it is maintained by a growing set of
+contributors including those two.
```
Comment (andreyorst) [inline doc/UltiSnips.txt]: As not native speaker, I may not understand "those two", but it's just me.

**Comment by SnowCone161803** [comment]
I'll make some of these changes tomorrow if you don't pick it up; I probably should have done that in the first place!

**Comment by SirVer** [comment]
@SnowCone161803 Awesome review! Thanks for that. I love your attention to detail :)

I will be pretty busy the rest of the week, so I will not be able to resolve your comments - feel free to do so if you find time.

**Comment by SirVer** [comment]
@SnowCone161803 Thanks for your contribution, Gordon! This is a great improvement.

@UltiBot merge

**Comment by SnowCone161803** [comment]
Thanks for all the feedback @SirVer ! I had fun!

**Pull request merged**